hack/make/build-deb

```diff
@@ -86,6 +86,16 @@ set -e
 		cat >> "$DEST/$version/Dockerfile.build" <<-EOF
 			RUN cp -aL hack/make/.build-deb debian
 			RUN { echo '$debSource (${debVersion}-0~${suite}) $suite; urgency=low'; echo; echo '  * Version: $VERSION'; echo; echo " -- $debMaintainer  $debDate"; } > debian/changelog && cat >&2 debian/changelog
+		EOF
+		# Remove the following case-based substitution when none of these are supported, and the TasksMax value is uncommented in docker.service
+		case "$version" in
+		debian-jessie|debian-wheezy|ubuntu-precise|ubuntu-trusty|ubuntu-wily)
+			;;
+		*)
+			echo "RUN sed -i -- 's/#TasksMax=infinity/TasksMax=infinity/' contrib/init/systemd/docker.service" >> "$DEST/$version/Dockerfile.build"
```

Comment:
If we can come up with a simple check for whether our current-environment
systemd is new enough, we could just add the check and the sed directly to
"override_dh_installinit" inside "debian/rules", and perform it directly on
the installed files in "debian/docker-engine/usr/.../docker.service"
instead of on the file within the repository (or perform the substitution
on the file directly before the current "dh_installinit" line on the file
in the repository, which occurs after the binary is built).